Bottom: 6a96759860c66540eda48f513970fb6a4d49827d
Top:    a6fef3b53b3e803ff31836e6a7fee319e48e52d0
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-04 20:07:21 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 31021c9..b01a5ae 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1101,7 +1101,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
